lib/CrEOF/Spatial/PHP/Types/AbstractGeometry.php

```diff
@@ -25,117 +26,141 @@
 
 use CrEOF\Spatial\Exception\InvalidValueException;
 use CrEOF\Spatial\PHP\Types\Geometry\GeometryInterface;
+use JsonSerializable;
 
 /**
- * Abstract geometry object for spatial types
+ * Abstract geometry object for spatial types.
  *
  * @author  Derek J. Lambert <dlambert@dereklambert.com>
- * @license http://dlambert.mit-license.org MIT
+ * @license https://dlambert.mit-license.org MIT
  */
-abstract class AbstractGeometry implements GeometryInterface
+abstract class AbstractGeometry implements GeometryInterface, JsonSerializable
 {
     /**
+     * Spatial Reference System Identifier.
+     *
      * @var int
      */
     protected $srid;
 
     /**
-     * @return array
+     * Spatial Reference System Identifier getter.
+     *
+     * @return int|null
      */
-    abstract public function toArray();
+    public function getSrid()
+    {
+        return $this->srid;
+    }
 
     /**
-     * @return string
+     * Specify data which should be serialized to JSON.
+     *
+     * @see https://php.net/manual/en/jsonserializable.jsonserialize.php
+     * @see https://github.com/creof/doctrine2-spatial/issues/140
+     *
+     * @return mixed data which can be serialized by <b>json_encode</b>,
+     *               which is a value of any type other than a resource
+     *
+     * @since 2.0.0.rc-1
      */
-    public function __toString()
+    public function jsonSerialize()
+    {
+        return [
+            'type' => $this->getType(),
+            'coordinates' => $this->toArray(),
+            'srid' => $this->getSrid(),
+        ];
+    }
+
+    /**
+     * Spatial Reference System Identifier fluent setter.
+     *
+     * @param mixed $srid Spatial Reference System Identifier
+     *
+     * @return self
+     */
+    public function setSrid($srid)
     {
-        $type   = strtoupper($this->getType());
-        $method = 'toString' . $type;
+        if (null !== $srid) {
+            $this->srid = (int) $srid;
+        }
 
-        return $this->$method($this->toArray());
+        return $this;
     }
 
     /**
+     * Convert this abstract geometry to an array.
+     *
+     * @return array
+     */
+    abstract public function toArray();
+
+    /**
+     * Convert this abstract geometry to a Json string.
+     *
      * @return string
      */
     public function toJson()
     {
+        $json = [];
         $json['type'] = $this->getType();
         $json['coordinates'] = $this->toArray();
-        return json_encode($json);
-    }
+        $json['srid'] = $this->getSrid();
 
-    /**
-     * @return null|int
-     */
-    public function getSrid()
-    {
-        return $this->srid;
+        return json_encode($json);
```

Comment:
`return json_encode($this);` same output because class implements `JsonSerializable`

Reply:
I added JsonSerializable because I read #194 , but (I'm not sure) I encountered an error in a specific case. Do you agree to wait this PR to be merged, then I will tried it.

Reply:
@Alexandre-T `JsonSerializable` is good. I just mention that no need to create array to generate json from `toJson` function.

following function will be enough because `JsonSerializable` feature in use.

```php
public function toJson()
{
    return json_encode($this);
}
```